libbeat/docs/outputconfig.asciidoc

```diff
@@ -521,6 +544,18 @@ The maximum number of files to save under <<path>>. When this number of files is
 oldest file is deleted, and the rest of the files are shifted from last to first. The default
 is 7 files.
 
+===== bulk_max_size
+
+The maximum number of events to bulk into batches. If the Beat sends single events, the events
+are collected into batches. If the Beat publishes a large batch of events (larger than the value
+specified by `bulk_max_size`), the batch is split.
+
+Specifying a larger batch size can improve performance by lowering the overhead of sending events. 
+However big batch sizes can also increase processing times.
+
+The default is 200.
```

Comment:
yes, 2048 is the default for all output plugins.

In libbeat we do have a generic publisher pipeline and output plugins responsible for actually sending the events. The buffering to generate bulk-events is done within the  internal publisher pipeline. As long as an output plugin does not disable bulk-events during setup, buffering will happen internally in libbeat. That is, this option basically exists for every single output plugin.

Just double-checked the code. It's actually the file output plugin which disabled bulk-events. All other outputs support bulk-events with default size of 2048 events, but elasticsearch sets default limit to 50. Sorry for the confusion.